src/app/modules/ix-table2/async-data-provider.ts

```diff
@@ -0,0 +1,41 @@
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { EmptyType } from 'app/enums/empty-type.enum';
+import { ArrayDataProvider } from 'app/modules/ix-table2/array-data-provider';
+
+export class AsyncDataProvider<T> extends ArrayDataProvider<T> {
+  private _subscription = new Subscription();
+  private _emptyType$ = new BehaviorSubject<EmptyType>(EmptyType.Loading);
+
+  get emptyType$(): Observable<EmptyType> {
```

Comment:
BehaviourSubject is already an observable. Can you use it directly and just mark property as `readonly`?